mne/io/meas_info.py

```diff
@@ -1917,25 +1963,31 @@ def anonymize_info(info, daysback=None, keep_his=False):
     default_desc = ("Anonymized using a time shift"
                     " to preserve age at acquisition")
 
-    # datetime object representing meas_date
-    meas_date_datetime = _stamp_to_dt(info['meas_date'])
-
-    if daysback is None:
-        delta_t = meas_date_datetime - default_anon_dos
+    # compute timeshift delta
+    if daysback is None and info['meas_date'] is None:
+        delta_t = datetime.timedelta(days=np.random.randint(365, 45 * 365))
```

Comment:
i would not do this. It means the file you generate will be different if you call the function twice. this is error prone. I see already a version control system commit a huge file for nothing. I would enforce daysback to be not None or set the date to 1970-01-01 as if date we removed.

Reply:
The problem with making `daysback` a required parameter is that it will encourage users to just pick a number (and likely the same number for all data). When in most cases people don't have longitudinal data and shouldn't need/want `daysback` at all. To properly use `daysback` the site needs to use a different number for each subject but the same number for datasets from the same subject.

The problem with picking a date for `meas_date is None` cases is that then we would always be using that same `daysback` which is just a constant timeshift of potentially identifiable information (birthday, device info, proc history etc)

In the latest round of edits. I just remove all the date/time info i know about if 'meas_date is None' I think this is robust in that you'll get the same `info` out each time and that it will be anonymous. I also don't think it loses any important time interval data.